src/_findErroneousUnionNode.ts

```diff
@@ -0,0 +1,78 @@
+import { Cube, LevelCoordinates } from "@activeviam/data-model";
+import {
+  findDescendant,
+  isMdxFunction,
+  Mdx,
+  MdxFunction,
+} from "@activeviam/mdx";
+
+import { _doesCrossjoinRepresentAnExpandedMember } from "./_doesCrossjoinRepresentAnExpandedMember";
+import { _doesCrossjoinYieldAllCombinationsOfMembers } from "./_doesCrossjoinYieldAllCombinationsOfMembers";
+
+/**
+ * The shape of an erroneous union node.
+ * Erroneous union nodes can lead to errors and must be cleaned up.
+ */
+export interface ErroneousUnionNode extends MdxFunction {
+  name: "Union";
+  arguments: (MdxFunction & {
+    name: "Crossjoin";
+  })[];
+}
+
+/**
+ * Returns whether `mdx` represents an erroneous union node.
+ */
+export function _findErroneousUnionNode(
+  mdx: Mdx,
+  { cube, levelsOnAxis }: { cube: Cube; levelsOnAxis: LevelCoordinates[] }
+): { path: (string | number)[]; match: ErroneousUnionNode } | undefined {
+  const result = findDescendant(mdx, (node) => {
+    if (!isMdxFunction(node, "union") || node.arguments.length < 2) {
+      // The erroneous node is an union function, with at least 2 arguments.
+      return false;
+    }
+
+    if (
+      !node.arguments.every((argument) => isMdxFunction(argument, "crossjoin"))
+    ) {
+      // All its arguments are crossjoins.
+      return false;
+    }
+
+    let numberOfCrossjoinsYieldingAllCombinationOfMembers = 0;
+    let numberOfCrossjoinsRepresentingAnExpandedMember = 0;
+
+    // This cast is safe. See early return above.
+    (node.arguments as MdxFunction[]).forEach((crossjoin) => {
```

Comment:
Using a `type guard function` in the above if statement to assert that arguments are functions would spare this cast.

Reply:
There already is a typeguard used here (`isMdxFunction`). But as it is used on an array, using `arr.every(typeguard)`, TypeScript cannot tell that `node.arguments` is an array of the type ensured by `isMdxFunction`.

I might be able to make it work without the case if I replace the `.forEach` by a for loop.

Reply:
Can't something like this make it work : 
```tsx
const isUnionNodeWithCrossJoinArguments = (
  node: Mdx
): node is MdxFunction & {
  arguments: [
    MdxFunction & { name: "crossjoin" },
    MdxFunction & { name: "crossjoin" }
  ];
}
```

Reply:
I could do it this way. And actually, this is exactly what I used to call `ErroneousUnionNode`. But I prefer the name you give it: it is more nature-oriented (less role-oriented).